### ISSUE TITLE
Structural Integrity Repaired More Than Original Value

### ISSUE THREAD

**Issue by SirMegaV** [MegaMek/mekhq#147]
Happening to small crafts, dropships and aeros (not sure about conventional and warships), after repairs, sometime the unit SI is more than it should be. Attached screen shot and save file.

Cannot replicate, but I suspect it has something to do with long SI repairs.

[StructuralIntegrityRepairedMoreThanOriginalValue.rar.gz](https://github.com/MegaMek/mekhq/files/451803/StructuralIntegrityRepairedMoreThanOriginalValue.rar.gz)

**Comment by kipstafoo** [comment]
I tried to replicate this on a dozen small craft and dropships over several repairs on each but it never went over the maximum.

**Comment by SirMegaV** [comment]
I tried doing it on purpose too but can never replicate. But it's
happening.  I think its a combination of conditions especially when bring
forward days are involved.

**Comment by Dylan-M** [comment]
@kipstafoo I wouldn't put any effort into this until after @Akjosch is done with his repair system re-write. In fact, assigning this (as yet another repair/refit bug) to him so he's aware.

**Comment by Tamren** [comment]
I get this bug with Sabre aerospace fighters. They normally have a stock SI of 11. However the SI has a tendency to climb exponentially between missions until it can reach 20000 SI or more, this rampantly inflates the BV of the Sabre and makes it virtually indestructible. This seems to be related to repairs because the longer the repair the more the SI inflates, even if the damage is just a few points of armour off the nose. In once extreme case I had to fix the avionics from scratch, using overtime this took my tech almost a week. This issue also happens sometimes with VTOLs.

I would consider this low priority because you can fix the SI simply by opening the GM tab and restoring each Sabre to normal condition. However I noticed that I can't open the damage edit tab after a battle for both aeros or VTOLs, or edit the damage through the GM tools in the hangar menu. That might be related.

**Comment by Dylan-M** [comment]
A version would be very important for this, even if it is a secondary report. Thanks.

https://github.com/MegaMek/megamek/wiki/Making-a-Bug-Report

**Comment by Tamren** [comment]
Right sorry, still figuring this github thing out. I can confirm this bug for MekHQ v42.

I am 99% certain that this has something to do with the repair system because ASF structural integrity numbers do not fluctuate unless I deploy them in a battle. When they return the amount of SI gain is related to how much time the repairs take. I just played a raid scenario where I deployed 6 Sabre ASFs, all of whom took minor armour damage of 10 points or less. When they returned and the repairs were completed each had doubled their SI from 11 to 22. Something that may be related is that I always send my ASFs into battle with external stores like rocket pods.

This is the last save file I have from before I noticed the SI problem and fixed it using the GM tools. All 6 ASFs are Sabre SB-27 light aeros. The three lowest have 704 SI each, and the worst one has 22416 SI.
[C66.zip](https://github.com/MegaMek/mekhq/files/792528/C66.zip)
